testing/regress/ecl/supercopy.ecl

```diff
@@ -88,116 +145,270 @@ rec supercopy(rec l) := TRANSFORM
   SELF.asSuperfile := l.asSuperfile;
 end;
 
-dst1 := NOFOLD(DATASET([{superFileName, superFileCopyName, 'mythor', False, 'Copy superfile as logical file:', ''}], rec));
-p1 := PROJECT(NOFOLD(dst1), supercopy(LEFT));
-c1 := CATCH(NOFOLD(p1), ONFAIL(TRANSFORM(rec,
-                                 SELF.result := 'Copy superfile as logical file: Fail',
+doSuperCopy(DATASET(rec) dst) := FUNCTION
+    p1a := PROJECT(NOFOLD(dst), supercopy(LEFT));
+    c1a := CATCH(NOFOLD(p1a), ONFAIL(TRANSFORM(rec,
+                                 SELF.result := dst[1].result + ' Fail',
                                  SELF.msg := FAILMESSAGE,
-                                 SELF.sourceFileName := superFileName,
-                                 SELF.destFileName := superFileCopyName,
-                                 SELF.destCluster := 'mythor',
-                                 SELF.asSuperfile := False
+                                 SELF.sourceFileName := dst[1].sourceFileName,
+                                 SELF.destFileName := dst[1].destFileName,
+                                 SELF.destCluster := dst[1].destCluster,
+                                 SELF.asSuperfile := dst[1].asSuperfile
```

Comment:
It's a bit odd to be passing in a dataset (with must have 1 row), when really regular parameters are wanted.
I think you could avoid by making both doSuperCopy and supercopy functions that take string/boolean parameters.
Then the callers wouldn't need to create a pseudo dataset to pass parameters in.